core/src/main/java/org/jdbi/v3/core/mapper/reflect/ReflectionMappers.java

```diff
@@ -115,6 +118,35 @@ public ReflectionMappers setCaseChange(UnaryOperator<String> caseChange) {
         return this;
     }
 
+    /**
+     * Set the strategy Jdbi uses for Java accessibility rules.
+     * The legacy default is to call {@code setAccessible(true)} in certain cases when we try to use a Constructor, Method, or Field.
+     * In the future, this default will be changed to a no-op, to better interact with the Java module system.
+     */
+    @Beta
+    public ReflectionMappers setAccessibleObjectStrategy(Consumer<AccessibleObject> makeAccessible) {
```

Comment:
this seems an incredible convoluted way to say "call setAccessible(true)" on any accessible object. Are there real use cases where people will register actual code that looks at the passed in Accessible object (which is a constructor or a method, so not even a class) and decide "I make this accessible but not that other thing")? 

This seems overkill. A simple boolean flag might suffice. or a `Function<Class<?>, Boolean>` that decides on a per-class base whether to make its members accessible or not.